pkg/session/session.go

```diff
@@ -110,18 +106,10 @@ func GetOrCreate(ctx context.Context, params *Params) (*Session, error) {
 
 	sessionKey := params.server + params.userinfo.Username() + params.datacenter
 	if cachedSession, ok := sessionCache[sessionKey]; ok {
-		logger = logger.WithValues("server", params.server, "datacenter", params.datacenter)
-		// if keepalive is enabled we depend upon roundtripper to reestablish the connection
-		// and remove the key if it could not
-		if params.feature.EnableKeepAlive {
-			return &cachedSession, nil
-		}
-		var err error
-		if ok, err = cachedSession.SessionManager.SessionIsActive(ctx); ok {
-			logger.V(2).Info("found active cached vSphere client session")
-			return &cachedSession, nil
-		}
-		logger.V(2).Error(err, "error checking if session is active")
+		// Since keepalive is always enabled, we return a cached session when available.
+		// To make sure that the cached session is still active, we depend on the roundtripper -- which, as configured
+		// in newClient(), removes the session from the cache if it dies.
+		return &cachedSession, nil
```

Comment:
I believe I've not changed anything here except for factoring out the KeepAlive check. Inactive sessions are being removed by the roundtripper as configured through a callback in `newClient()` below.

Reply:
There is a check for seeing if the retrieved session is active, I know it is a bit defensive but we haven't seen any scale problems yet with this. Was suggesting to add it back in case, but I see your point.

Reply:
Hope this works out fine! Happy to work on this again in case it needs some more safeguards.